atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweepTaskRunnerImpl.java

```diff
@@ -134,82 +132,89 @@ public SweepResults run(TableReference tableRef, int batchSize, @Nullable byte[]
         }
 
         startRow = MoreObjects.firstNonNull(startRow, PtBytes.EMPTY_BYTE_ARRAY);
-        RangeRequest rangeRequest = RangeRequest.builder().startRowInclusive(startRow).batchHint(batchSize).build();
+        RangeRequest rangeRequest = RangeRequest.builder()
+                .startRowInclusive(startRow)
+                .batchHint(batchSize)
+                .build();
 
-        long sweepTimestamp = getSweepTimestamp(sweepStrategy);
+        SweepStrategySweeper strategySweeper = getStrategySweeperFor(sweepStrategy);
 
-        ClosableIterator<RowResult<Value>> valueResults = ClosableIterators.emptyImmutableClosableIterator();
-        if (sweepStrategy == SweepStrategy.THOROUGH) {
-            valueResults = keyValueService.getRange(tableRef, rangeRequest, sweepTimestamp);
-        }
+        long sweepTimestamp = strategySweeper.getSweepTimestamp();
 
-        ClosableIterator<RowResult<Set<Long>>> rowResults =
-                keyValueService.getRangeOfTimestamps(tableRef, rangeRequest, sweepTimestamp);
+        ClosableIterator<RowResult<Value>> valueResults = strategySweeper.getValues(tableRef, rangeRequest, sweepTimestamp);
+        ClosableIterator<RowResult<Set<Long>>> rowResults = strategySweeper.getCellTimestamps(tableRef, rangeRequest, sweepTimestamp);
```

Comment:
probably want to ensure we close both of these via try-with-resources in case `getCellTimestamps` throws and `valueResults` doesn't get closed.

```
    try (ClosableIterator<RowResult<Value>> valueResults = strategySweeper.getValues(tableRef, rangeRequest, sweepTimestamp);
        ClosableIterator<RowResult<Set<Long>>> rowResults = strategySweeper.getCellTimestamps(tableRef, rangeRequest, sweepTimestamp)) {
```